GoogleForCreators/web-stories-wp
# Saved Page Templates: fix repeated image uploads

## Context



As per #10787, the list of saved templates in the `LibraryProvider` is not updated after uploading the featured image for a saved page template.

That causes the repeated image uploads due to stale data.

## Summary



Fix repeated image uploads by uploading state in the provider after uploading the featured image.

## Relevant Technical Choices



* Added a new `updateSavedTemplate` callback to allow updating a single saved template in the existing state.

## To-do



* Maybe add tests? Not sure how best to test this.

## User-facing changes



## Testing Instructions






This PR can be tested by following these steps:

1. Go the Templates tab
2. Save current page as template
3. Observe HTTP requests to `wp-json/web-stories/v1/media`, notice image being uploaded
4. Go to another tab, e.g. Media
5. Switch back to Templates tab
6. There should be no new HTTP requests due to media uploads

## Reviews

### Does this PR have a security-related impact?



No

### Does this PR change what data or activity we track or use?



No

### Does this PR have a legal-related impact?



No

## Checklist




---



Fixes #10787
